Automattic/jetpack
# Bugfix: font sizes in payment-buttons taking precedence over contained buttons

Fixes #26748

#### Changes proposed in this Pull Request:

The payment-buttons typography feature introduced in #26580 had a bug
which meant that the font size in the payment-buttons block was
overriding font sizes set on the buttons when a preset font size was
being used. This bug occured when viewing the site, but not when
editing.
    
The bug was caused by the view CSS selector not being quite specific
enough.
    
The font size of the button blocks when using a preset works by adding a
has-{presetName}-font-size class to the element in question, where
presetName varies depending upon the theme. To fix this bug whenever a
preset font size is used a class called has-custom-font-size is added.
This gives a known stable class name which is targeted with CSS.

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:







 1. Apply patch / add a JN site etc
 2. Open Gutenberg
 3. Add 4 payment buttons.
 4. Update the size in the right setting panel. Make one small, one XL, one a set font size e.g. 15px and don't give any font-size to the last
 5. Select the payment buttons block and change the size to small 
 6. Save page
 7. Check it in frontend, it should look like in the editor: two small buttons at either end and in the middle an XL button and small looking one with a font-size of 15px
